src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -97,22 +100,29 @@ namespace AppInstaller::CLI::Workflow
                 installerArgs += ' ' + installerSwitches.at(ManifestInstaller::InstallerSwitchType::Language);
             }
 
-            // Construct install location arg if necessary.
-            if (context.Args.Contains(Execution::Args::Type::InstallLocation) && installerSwitches.find(ManifestInstaller::InstallerSwitchType::InstallLocation) != installerSwitches.end())
-            {
-                installerArgs += ' ' + installerSwitches.at(ManifestInstaller::InstallerSwitchType::InstallLocation);
-            }
-
             // Construct log path arg.
             if (installerSwitches.find(ManifestInstaller::InstallerSwitchType::Log) != installerSwitches.end())
             {
                 installerArgs += ' ' + installerSwitches.at(ManifestInstaller::InstallerSwitchType::Log);
             }
 
-            // Construct custom arg.
-            if (installerSwitches.find(ManifestInstaller::InstallerSwitchType::Custom) != installerSwitches.end())
+            if (isUpdate && installerSwitches.find(ManifestInstaller::InstallerSwitchType::Update) != installerSwitches.end())
+            {
+                installerArgs += ' ' + installerSwitches.at(ManifestInstaller::InstallerSwitchType::Update);
+            }
+            else
+            {
+                // Construct custom arg.
+                if (installerSwitches.find(ManifestInstaller::InstallerSwitchType::Custom) != installerSwitches.end())
+                {
+                    installerArgs += ' ' + installerSwitches.at(ManifestInstaller::InstallerSwitchType::Custom);
+                }
+            }
+
+            // Construct install location arg if necessary.
+            if (!isUpdate && context.Args.Contains(Execution::Args::Type::InstallLocation) && installerSwitches.find(ManifestInstaller::InstallerSwitchType::InstallLocation) != installerSwitches.end())

```

Comment:
>InstallLocation [](start = 74, length = 15)

Will the "remove first" update flow just be uninstall + install?  Because I could see that potentially needing the install location here.

Reply:
Would be better handled by upgrade simply removing the flag so that every switch could benefit at the same time.

---
In reply to: [499903564](https://github.com/microsoft/winget-cli/pull/588#discussion_r499903564) [](ancestors = 499903564)